### PR TITLE
Add how to add animation call method tracks/keys through code.

### PR DESCRIPTION
**Pull request by RancidMilkGames**
I added how the `key` Variant for adding an animation key to a call method animation track needs to be a dictionary, what the dictionary's key names are and their expected values, and a sample snippet of doing it in both GDScript and C#.

This will be my first contribution, so I apologize if I didn't do something correctly. I read through the contribution process, how you're supposed to write, and peeked around for similar preexisting ways things have been done in the documentation to avoid what pitfalls I could. I'm happy to change whatever if something feels off or is wrong.

### CODE REVIEW

**Review by mhilbrunner — APPROVED** [review]
The content here looks good to me.
1. The C# needs a quick glance by someone more versed there (@raulsntos maybe? :))
2. I'm not sure if this should not largely be in the class reference for `Animation.track_insert_key()` (if shortened a bit), and with this then pointing there. Especially something like the table for the passed `dict` is purely "how to use this single method", which IMO is always class reference territory. Opinions?

File: tutorials/animation/animation_track_types.rst [hunk 1]
```diff
@@ -57,6 +57,80 @@ inspector dock. There, you can change the method to call. If you expand the
...
+        var methodDictionary = new Godot.Collections.Dictionary
+        {
+            { "method", "Jump" },
+            { "args", new Godot.Collections.Array{jumpVelocity, multiplier} }
+        };
+
+        // Set scene-tree path to node with target method
+        animation.TrackSetPath(trackIndex, ".");
+        // Add the dictionary as the animation method track's key
+        animation.TrackInsertKey(trackIndex, 0.2, methodDictionary, 0);
```
Comment (raulsntos) [inline tutorials/animation/animation_track_types.rst:124]: The GDScript example uses `0.6` for the `time` parameter, is there a reason for this difference?

File: tutorials/animation/animation_track_types.rst [hunk 2]
```diff
@@ -57,6 +57,80 @@ inspector dock. There, you can change the method to call. If you expand the
...
+    }
+
+
+    // The target method that will be called from the animation
+    public void Jump(float jumpVelocity, float multiplier)
```
Comment (raulsntos) [inline tutorials/animation/animation_track_types.rst:129]: It doesn't look like this needs to be public.

```suggestion
    private void Jump(float jumpVelocity, float multiplier)
```

File: tutorials/animation/animation_track_types.rst [hunk 3]
```diff
@@ -57,6 +57,80 @@ inspector dock. There, you can change the method to call. If you expand the
...
+        var animation = animationPlayer.GetAnimation("idle");
+        // Get or create the target method's animation track
+        var trackIndex = animation.AddTrack(Animation.TrackType.Method);
+        // Make the arguments for the target method jump()
+        var jumpVelocity = -400.0;
+        var multiplier = GD.RandRange(.8, 1.2);
+        // Get or create a dictionary with the target method's name and arguments
+        var methodDictionary = new Godot.Collections.Dictionary
+        {
+            { "method", "Jump" },
```
Comment (raulsntos) [inline tutorials/animation/animation_track_types.rst:117]: We want to encourage users to use the cached StringNames when possible.

```suggestion
            { "method", MethodName.Jump },
```

File: tutorials/animation/animation_track_types.rst [hunk 4]
```diff
@@ -57,6 +57,80 @@ inspector dock. There, you can change the method to call. If you expand the
...
+            { "args", new Godot.Collections.Array{jumpVelocity, multiplier} }
```
Comment (raulsntos) [inline tutorials/animation/animation_track_types.rst:118]: Just to be consistent with the code style we use everywhere else.

```suggestion
            { "args", new Godot.Collections.Array { jumpVelocity, multiplier } }
```

File: tutorials/animation/animation_track_types.rst [hunk 5]
```diff
@@ -57,6 +57,80 @@ inspector dock. There, you can change the method to call. If you expand the
...
+    func jump(jump_velocity, multiplier):
+        velocity.y = jump_velocity * multiplier
+
+ .. code-tab:: csharp
+
+    // Create a call method track
+    public void CreateAnimationTrack()
+    {
+        // Get reference to the AnimationPlayer
+        var animationPlayer = GetNode("AnimationPlayer") as AnimationPlayer;
```
Comment (raulsntos) [inline tutorials/animation/animation_track_types.rst:106]: We tend to prefer using the generic version of this method in the documentation.

```suggestion
        var animationPlayer = GetNode<AnimationPlayer>("AnimationPlayer");
```

**Comment by RancidMilkGames** [comment]
Thanks, can I ask if @mhilbrunner or anyone has some further thought about their second point here before I resubmit with the changes?

> The content here looks good to me.
> 
>     1. The C# needs a quick glance by someone more versed there (@raulsntos maybe? :))
> 
>     2. I'm not sure if this should not largely be in the class reference for `Animation.track_insert_key()` (if shortened a bit), and with this then pointing there. Especially something like the table for the passed `dict` is purely "how to use this single method", which IMO is always class reference territory. Opinions?

File: tutorials/animation/animation_track_types.rst (hunk 1 above)
Comment (RancidMilkGames) [reply tutorials/animation/animation_track_types.rst:124]: No, I have changed it back for the fixes commit I'm about to make, with the other brought up changes. Sorry, I'm guessing that happened when I was confirming it ran.

**Comment by RancidMilkGames** [comment]
Hi, I believe I got all the changes. Punctuation in the comments. inconsistency between language examples. Public to Private. Cached StringName instead of String. Generic version of GetNode.

File: tutorials/animation/animation_track_types.rst [hunk 6]
```diff
@@ -57,6 +57,80 @@ inspector dock. There, you can change the method to call. If you expand the
...
+    // Create a call method track.
+    public void CreateAnimationTrack()
+    {
+        // Get reference to the AnimationPlayer.
+        var animationPlayer = GetNode<AnimationPlayer>("AnimationPlayer");
+        // Get or create the animation the target method will be called from.
+        var animation = animationPlayer.GetAnimation("idle");
+        // Get or create the target method's animation track.
+        var trackIndex = animation.AddTrack(Animation.TrackType.Method);
+        // Make the arguments for the target method jump().
```
Comment (raulsntos) [inline tutorials/animation/animation_track_types.rst:111]: Looks like I missed this before. The comment references the `Jump` method so it should match the capitalization.

```suggestion
        // Make the arguments for the target method Jump().
```

**Review by raulsntos — APPROVED** [review]
The C# documentation looks good to me, minus a nitpick.

File: tutorials/animation/animation_track_types.rst (hunk 6 above)
Comment (RancidMilkGames) [reply tutorials/animation/animation_track_types.rst]: Gotcha! I have made the change and pushed it

**Review by mhilbrunner — APPROVED** [review]

**Review by skyace65 — APPROVED** [review]

**Comment by skyace65** [comment]
Thanks! And congrats on your first merged PR! In the future please keep your PRs to one commit, I did a squash and merge for this one but just keep that in mind for the future. If you don't know how to keep everything as 1 commit we have a guide [here](https://docs.godotengine.org/en/stable/contributing/workflow/pr_workflow.html#modifying-a-pull-request).

**Comment by RancidMilkGames** [comment]
@skyace65
Yay!! I apologize, I think I remember seeing a Github hint that says you can do it the way I did, and I was uncertain how that would behave when merged. I should have thought that there might be a section in the docs for that. Thanks, and from now on I will make sure to do it the way you just linked.

**Pull request merged**